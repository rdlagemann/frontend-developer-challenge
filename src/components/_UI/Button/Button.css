.Button {
  width: 160px;
  border-radius: 4px;
  background: white;
  margin: 0 auto;
  color: var(--gray);
  font-size: 14px;
  height: 40px;
  font-weight: bold;
}

.Button_primary {
  color: var(--primary);
}

.Button:hover {
  cursor: pointer;
}

.Button-link {
  text-decoration: none;
  display: flex;
  justify-content: center;
  align-items: center;
}
