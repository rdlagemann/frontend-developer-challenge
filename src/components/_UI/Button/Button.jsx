import React from 'react'
import './Button.css'
import cx from 'classnames'

export default function Button({
  className,
  label,
  children,
  tag = '',
  color = 'primary',
  sm = false,
  outline = false,
  ...restProps
}) {
  const baseClass = cx('Button', className, {
    Button_primary: color === 'primary',
    Button_secondary: color === 'secondary',
    Button_outline: outline,
    Button_small: sm
  })

  if (tag === 'a') {
    return (
      <a className={cx(baseClass, 'Button-link')} {...restProps}>
        {children || label}
      </a>
    )
  }
  return (
    <button className={baseClass} {...restProps}>
      {children || label}
    </button>
  )
}
