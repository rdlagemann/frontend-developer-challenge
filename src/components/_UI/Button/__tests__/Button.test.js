import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import Button from '..'

describe('Button', () => {
  it('renders', () => {
    const output = create(<Button />)
    expect(output).toMatchSnapshot()
  })
})

describe('Button ', () => {
  it('should contains a label', () => {
    const label = 'Click me'
    const button = shallow(<Button label={label} />)
    expect(button.contains(label)).toBe(true)
  })
})
