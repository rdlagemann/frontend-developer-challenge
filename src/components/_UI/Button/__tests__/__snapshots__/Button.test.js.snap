// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders 1`] = `
<button
  className="Button Button_primary"
/>
`;
